### PR TITLE
Improve ASCAT

### PR DESCRIPTION
**Pull request by malinlarsson**
- Ascat has been updated to version 2.5.2.
- The update fixed problems on the X-chromosome in male samples.
- Also, the process of converting AlleleCounts to LogR values and running Ascat is updated based on the code in https://github.com/cancerit/ascatNgs/blob/dev/perl/share/ascat/runASCAT.R.
- Input parameters enabled to override Ascats calculated values for purity and ploidy (set with --ascat_purity and --ascat_ploidy)
PR checklist

- [x] This comment contains a description of changes (with reason)
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If necessary, also make a PR on the nf-core/sarek branch on the nf-core/test-datasets repo
- [ ] Ensure the test suite passes (nextflow run . -profile test,docker).
- [ ] Make sure your code lints (nf-core lint .).
- [x] Documentation in docs is updated
- [ ] CHANGELOG.md is updated
- [ ] README.md is updated

EDIT: rearrange text

### CODE REVIEW

File: docs/install_bianca.md [hunk 1]
```diff
@@ -146,12 +146,32 @@ So every member of the project who wants to use nf-core/sarek will need to do:
 > ln -s /castor/project/proj_nobackup/sarek/default sarek
 ```
 
+Singularity images for Sarek are available on Uppmax in /sw/data/uppnex/ToolBox/sarek. Sometimes nextflow needs write access to the image folder, and if so the images needs to be copied to a location with write permission, for example in a subfolder of your project folder.
```
Comment (MaxUlysse) [inline docs/install_bianca.md]:
```suggestion
Singularity images for Sarek are available on Uppmax in `/sw/data/uppnex/ToolBox/sarek`.
Sometimes Nextflow needs write access to the image folder, and if so the images needs to be copied to a location with write permission, for example in a subfolder of your project folder.
```

File: docs/install_bianca.md [hunk 2]
```diff
@@ -146,12 +146,32 @@ So every member of the project who wants to use nf-core/sarek will need to do:
 > ln -s /castor/project/proj_nobackup/sarek/default sarek
 ```
 
+Singularity images for Sarek are available on Uppmax in /sw/data/uppnex/ToolBox/sarek. Sometimes nextflow needs write access to the image folder, and if so the images needs to be copied to a location with write permission, for example in a subfolder of your project folder. 
+```bash
```
Comment (MaxUlysse) [inline docs/install_bianca.md]:
```suggestion

```bash
```

File: docs/install_bianca.md [hunk 3]
```diff
@@ -146,12 +146,32 @@ So every member of the project who wants to use nf-core/sarek will need to do:
...
+#Create a folder for the singularity images somewhere in your project:
+mkdir sarek_simg
+
+#Copy the relevant singularity image from the write protected folder on Uppmax to the fodler where you have write permission:
```
Comment (MaxUlysse) [inline docs/install_bianca.md]:
```suggestion
#Copy the relevant singularity image from the write protected folder on Uppmax to the folder where you have write permission:
```

File: bin/run_ascat.r [hunk 4]
```diff
@@ -10,19 +10,30 @@ if(length(args)<6){
     tumorname = args[5]
     baseDir = args[6]
     gcfile = args[7]
+    gender = args[8]
 }
 
-source(paste(baseDir,"/scripts/ascat.R", sep=""))
+#source(paste(baseDir,"/scripts/ascat.R", sep=""))
```
Comment (MaxUlysse) [inline bin/run_ascat.r]:
```suggestion
```

Comment (MaxUlysse) [reply bin/run_ascat.r]: You could even remove the `ascat.R` file from `bin`

File: bin/run_ascat.r [hunk 5]
```diff
@@ -10,19 +10,30 @@ if(length(args)<6){
...
+library(ASCAT)
 
 if(!require(RColorBrewer)){
     source("http://bioconductor.org/biocLite.R")
     biocLite("RColorBrewer", suppressUpdates=TRUE, lib="$baseDir/scripts")
     library(RColorBrewer)
-}
+}ls
```
Comment (MaxUlysse) [inline bin/run_ascat.r]: Just wondering about this `ls`

**Comment by MaxUlysse** [comment]
Amazing work.

I suggested some changes to fix the Markdown Linting that you can accept and batch commit from the `Files changed` tab.

Apart from that I have just one issue, we're not using submodules for the test-datasets and configs repo, as they are most of the time just pulled from internet when needed.
We're trying to improve the nf-core download command to include all that: https://nf-co.re/tools#downloading-pipelines-for-offline-use so I would remove the related `.gitmodules` and the repos as well.

File: bin/run_ascat.r (hunk 4 above)
Comment (MaxUlysse) [reply bin/run_ascat.r]: For that I guess you would need to be in your own branch (same for the modules stuff).
I'll just remove the file and then commit the change.
You might have to pull your branch if you accepted my suggestion.

File: bin/run_ascat.r (hunk 5 above)
Comment (MaxUlysse) [reply bin/run_ascat.r]:
```suggestion
}
```

File: main.nf [hunk 6]
```diff
@@ -1440,10 +1440,11 @@ process Ascat {
     when: 'ascat' in tools
 
     script:
+    gender = genderMap[idPatient]
     """
     # get rid of "chr" string if there is any
-    for f in *BAF *LogR; do sed 's/chr//g' \$f > tmpFile; mv tmpFile \$f;done
-    Rscript ${workflow.projectDir}/bin/run_ascat.r ${bafTumor} ${logrTumor} ${bafNormal} ${logrNormal} ${idSampleTumor} ${baseDir} ${acLociGC}
+    #for f in *BAF *LogR; do sed 's/chr//g' \$f > tmpFile; mv tmpFile \$f;done
```
Comment (MaxUlysse) [inline main.nf]: I can see that you commented this line.
If so, we can probably get rid of it and of the one on top as well?

File: bin/convertAlleleCounts.r [hunk 7]
```diff
@@ -2,6 +2,8 @@
 # Description:
 # R-script for converting output from AlleleCount to BAF and LogR values.
 #
+# Testing new version of the script
```
Comment (MaxUlysse) [inline bin/convertAlleleCounts.r]:
```suggestion
```
I think we can remove that comment ;-)

Comment (malinlarsson) [reply bin/convertAlleleCounts.r]: Yes

**Pull request merged**